plugins/org.eclipse.n4js.xpect/src/org/eclipse/n4js/xpect/common/SafeURIFileCollector.java

```diff
@@ -0,0 +1,68 @@
+package org.eclipse.n4js.xpect.common;
+
+import java.io.File;
+import java.util.Collection;
+import java.util.List;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.xpect.runner.XpectTestFiles;
+import org.eclipse.xpect.runner.XpectTestFiles.XpectTestFileCollector;
+import org.eclipse.xtext.util.UriExtensions;
+
+import com.google.common.collect.Lists;
+
+/**
+ * Customization that ensures that all produced file URIs do have an empty authority.
+ */
+public class SafeURIFileCollector extends XpectTestFileCollector {
+
+	private final UriExtensions uriExtensions = new UriExtensions();
+
+	/** Constructor */
+	public SafeURIFileCollector(Class<?> owner, XpectTestFiles ctx) {
+		super(owner, ctx);
+	}
+
+	@Override
+	protected URI createURI(File file) {
+		throw new UnsupportedOperationException("Never called by super impl");
```

Comment:
@szarnekow Why not just create a file URI, but with empty authority (corresponding to other overrides).

Reply:
Since it was never called as far as I could tell, I wanted to add a tripwire.